homeassistant/components/fan/knx.py

```diff
@@ -0,0 +1,142 @@
+"""
+Support for KNX/IP fans.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/fan.knx/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.knx import DATA_KNX, ATTR_DISCOVER_DEVICES
+from homeassistant.components.fan import (
+    PLATFORM_SCHEMA, FanEntity, SUPPORT_SET_SPEED)
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+CONF_SPEED_ADDRESS = 'speed_address'
+CONF_SPEED_STATE_ADDRESS = 'speed_state_address'
+
+DEFAULT_NAME = 'KNX Fan'
+DEPENDENCIES = ['knx']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SPEED_ADDRESS): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_SPEED_STATE_ADDRESS): cv.string,
+})
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_devices,
+                               discovery_info=None):
+    """Set up fan(s) for KNX platform."""
+    if discovery_info is not None:
+        async_add_devices_discovery(hass, discovery_info, async_add_devices)
+    else:
+        async_add_devices_config(hass, config, async_add_devices)
+
+
+@callback
+def async_add_devices_discovery(hass, discovery_info, async_add_devices):
+    """Set up fans for KNX platform configured via xknx.yaml."""
+    entities = []
+    for device_name in discovery_info[ATTR_DISCOVER_DEVICES]:
+        device = hass.data[DATA_KNX].xknx.devices[device_name]
+        entities.append(KNXFan(hass, device))
+    async_add_devices(entities)
+
+
+@callback
+def async_add_devices_config(hass, config, async_add_devices):
+    """Set up fan for KNX platform configured within plattform."""
+    import xknx
+    fan = xknx.devices.Fan(
+        hass.data[DATA_KNX].xknx,
+        name=config.get(CONF_NAME),
+        group_address_speed=config.get(CONF_SPEED_ADDRESS),
+        group_address_speed_state=config.get(CONF_SPEED_STATE_ADDRESS))
+    hass.data[DATA_KNX].xknx.devices.add(fan)
+    async_add_devices([KNXFan(hass, fan)])
+
+
+class KNXFan(FanEntity):
+    """Representation of a KNX fan."""
+
+    def __init__(self, hass, device):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.